File: spring-web-java/src/main/java/spring/web/java/domain/Member.java
package spring.web.java.domain;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import spring.web.java.constant.Role;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Member extends BaseDomain {

    @Id
    @GeneratedValue
    @Column(name = "member_id")
    private Long id;
    private String account;
    private String password;
    private String name;

    @Enumerated(EnumType.STRING)
    private Role role;

    @Embedded
    private Address address;

    @OneToMany(mappedBy = "member")
    private List<Order> orders = new ArrayList<>();

    public static Member create(String account, String password, String name, Role role, Address address) {
        Member member = new Member();

        member.account = account;
        member.password = password;
        member.name = name;
        member.role = role;
        member.address = address;

        return member;
    }

    public void update(String name, Role role, Address address) {
        this.name = name;
        this.role = role;
        this.address = address;
    }
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/Member.kt
package spring.web.kotlin.domain

import jakarta.persistence.*
import spring.web.kotlin.constant.Role

@Entity
class Member private constructor(
    @Id
    @GeneratedValue
    @Column(name = "member_id")
    val id: Long? = null,
    val account: String,
    val password: String,
    var name: String,

    @Enumerated(EnumType.STRING)
    var role: Role,

    @Embedded
    var address: Address
) {
    companion object {
        fun create(account: String, password: String, name: String, role: Role, address: Address) =
            Member(
                account = account,
                password = password,
                name = name,
                role = role,
                address = address
            )
    }

    fun update(name: String, role: Role, address: Address) {
        this.name = name
        this.role = role
        this.address = address
    }
}